spark3/src/main/java/org/apache/iceberg/spark/source/SparkBatchScan.java

```diff
@@ -158,8 +166,10 @@ public PartitionReaderFactory createReaderFactory() {
 
     boolean batchReadsEnabled = batchReadsEnabled(allParquetFileScanTasks, allOrcFileScanTasks);
 
+    boolean hasTimestampWithoutZone = SparkUtil.hasTimestampWithoutZone(expectedSchema);
+
     boolean readUsingBatch = batchReadsEnabled && hasNoDeleteFiles && (allOrcFileScanTasks ||
-        (allParquetFileScanTasks && atLeastOneColumn && onlyPrimitives));
+            (allParquetFileScanTasks && atLeastOneColumn && onlyPrimitives && !hasTimestampWithoutZone));
```

Comment:
We don't need the !hasTimestampWithoutZone here right? for batch reading? Or did we address this and I just forgot

Reply:
I have removed this check